src/script/script.h

```diff
@@ -508,8 +508,7 @@ class CScript : public CScriptBase
             return false;
 
         // Read instruction
-        if (end() - pc < 1)
-            return false;
```

Comment:
If it's not a valid iterator, just subtracting it from end() is probably undefined behaviour.